### ISSUE TITLE
Allow further installation languages (currently only English). Disk encryption password may not be entered correctly when the keyboard language is changed.

### ISSUE THREAD

**Issue by Wikinaut** [QubesOS/qubes-issues#3785]
### Qubes OS version:

Qubes R4.0

### Affected component(s): 

* installation
* harddisk encryption password

---

### Steps to reproduce the behavior:

R4.0 installation allows only a couple of English language on the first screen.

### Expected behavior:

Installation should also allowed further languages.

When having only English like at the moment. you cannot use non-Ascii-Characters for the disk encryption password, because it might happen, that you cannot enter the password later when you changed the keyword layout to German, for example., which keyboards have a different layount and you run into the "Y/Z" problem ("Y" and "Z" keys are exchanged on German vs. English keyboards).

### General notes:

---

**Comment by marmarek** [comment]
Language and keyboard layout are separate settings. You can choose any keyboard layout (including non-English ones). And using different keyboard layout finally works in Qubes 4.0 (https://github.com/QubesOS/qubes-issues/issues/3234). When you enter disk passphrase, there is clear indication what keyboard layout is chosen, If you want different one, set it before setting disk passphrase.

**Comment by marmarek** [comment]
How exactly have you configured keyboard layout? Try this:
1. Select "Keyboard" in main installer screen.
2. Add German keyboard
3. Move it to first position on that list (this will make it default keyboard layout)

**Comment by Wikinaut** [comment]
I did this already during two installations. On my Samsung Laptop, it seems to be okay, on my Intel NUC I have the described issue, which is reproducible.

Ad step 3 (above): in one test installation, I then removed the English keyboard (which then was on the second position).

Let me know, if you want me to do another test.

**Comment by marmarek** [comment]
Are you using final R4.0 image, or release candidate?

**Comment by marmarek** [comment]
Do you mean the same steps on different hardware leads to different results?
Is it that on PS2 (internal) keyboard it's ok, but on USB (external) keyboard it's wrong?

**Comment by Wikinaut** [comment]
Final R4.0

**Comment by Wikinaut** [comment]
Have you fixed this in the last update? It looks so to me.

**Comment by marmarek** [comment]
I guess regenerating initramfs (caused by kernel update) might fixed this. But it is still a mystery for me why it didn't worked for you with USB keyboard, but at the same time worked with internal one...

**Comment by Wikinaut** [comment]
As my problem appears to be "solved", I am closing it. Please reopen if needed by the team.

**Comment by Wikinaut** [comment]
@marmarek Please allow me to update my report regarding the Keyboad "issue", because I know now exactly the reason. The problem comes up in the following step during installation.

1. Installer Boot from USB, First stage of installation
2. (EN= Language, there is only English and En-sublanguages)
3. I add via "+" keyboard=GERMAN
4. I remove via "-" keyboard English
5. I set up a LUKS passphrase with an "Z" (German keyboard is "QWERTZ")
6. In the next boot - during the installation - I have to enter my passphrase for the very first time, and only then, it is still ENGLISH (keyboard "QWERTY") so that I have to take the YZ-swap into consideration when entering the LUKS passphrase.
7. Second stage of installation goes on
8. Qubes is installed.
9. From now on, all boots into the fully installed Qubes are correctly using my GERMAN keyboard layout.

So, in short, the very only time is number 6 - between stage 1 and 2 during the installation, when the user has to consider a (possible) swap of characters!

I hope you understand what I want to say: my issue is only an issue between 1st and 2nd isntallation stage, in other words: only the first passphrase goes via the ENGLISH keyboard layout, and users must consider a "swap" of YZ (and perhaps other characters, depending on their chosen keyboard layout).
